seatunnel-apis/seatunnel-api-spark/src/main/java/org/apache/seatunnel/spark/batch/SparkBatchExecution.java

```diff
@@ -0,0 +1,132 @@
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.seatunnel.spark.batch;
+
+import org.apache.seatunnel.common.config.CheckResult;
+import org.apache.seatunnel.common.config.ConfigRuntimeException;
+import org.apache.seatunnel.env.Execution;
+import org.apache.seatunnel.spark.BaseSparkSink;
+import org.apache.seatunnel.spark.BaseSparkSource;
+import org.apache.seatunnel.spark.BaseSparkTransform;
+import org.apache.seatunnel.spark.SparkEnvironment;
+
+import org.apache.seatunnel.shade.com.typesafe.config.Config;
+import org.apache.seatunnel.shade.com.typesafe.config.ConfigFactory;
+
+import org.apache.spark.sql.Dataset;
+import org.apache.spark.sql.Row;
+
+import java.util.List;
+
+public class SparkBatchExecution implements Execution<SparkBatchSource, BaseSparkTransform, SparkBatchSink> {
+
+    public static final String SOURCE_TABLE_NAME = "source_table_name";
+    public static final String RESULT_TABLE_NAME = "result_table_name";
+
+    public static void registerTempView(String tableName, Dataset<Row> ds) {
+        ds.createOrReplaceGlobalTempView(tableName);
+    }
+
+    public static void registerInputTempView(BaseSparkSource<Dataset<Row>> source, SparkEnvironment environment) {
+        Config config = source.getConfig();
+        if (config.hasPath(SparkBatchExecution.RESULT_TABLE_NAME)) {
+            String tableName = config.getString(SparkBatchExecution.RESULT_TABLE_NAME);
+            registerTempView(tableName, source.getData(environment));
+        } else {
+            throw new ConfigRuntimeException(
+                    "Plugin[" + source.getClass().getName() + "] must be registered as dataset/table, please set \"result_table_name\" config");
+        }
+    }
+
+    public static Dataset<Row> transformProcess(SparkEnvironment environment, BaseSparkTransform transform, Dataset<Row> ds) {
+        Dataset<Row> fromDs;
+        Config config = transform.getConfig();
+        if (config.hasPath(SparkBatchExecution.SOURCE_TABLE_NAME)) {
+            String sourceTableName = config.getString(SparkBatchExecution.SOURCE_TABLE_NAME);
+            fromDs = environment.getSparkSession().read().table(sourceTableName);
+        } else {
+            fromDs = ds;
+        }
+        return transform.process(fromDs, environment);
+    }
+
+    public static void registerTransformTempView(BaseSparkTransform transform, Dataset<Row> ds) {
+        Config config = transform.getConfig();
+        if (config.hasPath(SparkBatchExecution.RESULT_TABLE_NAME)) {
+            config.getString(SparkBatchExecution.RESULT_TABLE_NAME);
+            registerTempView(RESULT_TABLE_NAME, ds);
```

Comment:
there is a bug about tableName.

Reply:
[7402da4](https://github.com/apache/incubator-seatunnel/pull/1188/commits/7402da4b0359459912acdd72bf7714491bf89f99) had been resolved